membership-attribute-service/app/controllers/AccountController.scala

```diff
@@ -41,7 +42,12 @@ class AccountController(commonActions: CommonActions, override val controllerCom
   implicit val executionContext: ExecutionContext= controllerComponents.executionContext
   lazy val authenticationService: AuthenticationService = IdentityAuthService
 
-   def cancelSubscription [P <: SubscriptionPlan.AnyPlan : SubPlanReads] = BackendFromCookieAction.async { implicit request =>
+  def subscriptionSelector[P <: SubscriptionPlan.AnyPlan](subscriptionNameOption: Option[memsub.Subscription.Name], messageSuffix: String)(subscriptions: List[Subscription[P]]): String \/ Subscription[P] = subscriptionNameOption match {
+    case Some(subName) => subscriptions.find(_.name == subName) \/> s"$subName was not a subscription for $messageSuffix"
+    case None => subscriptions.headOption \/> s"No current subscriptions for $messageSuffix"
```

Comment:
Bye bye pervasive .headOption 🎉 I'm guessing once all endpoints have been changed in the client, we can remove this one too?

Reply:
correct (we would refactor out the `Option` of the parameter too)